src/challenge_package/src/move_action_client.cpp

```diff
@@ -0,0 +1,98 @@
+#include <memory>
+#include <chrono>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp_action/rclcpp_action.hpp"
+#include "challenge_package/action/move.hpp"
+
+using move_action = challenge_package::action::Move;
+using goal_handle_move = rclcpp_action::ClientGoalHandle<move_action>;
+
+class MoveActionClient : public rclcpp::Node 
+{
+public:
+  MoveActionClient()
+  : Node("move_action_client")
+  {
+    this->client_ptr = rclcpp_action::create_client<move_action>(
```

Comment:
In C++ you don't really use this, it's just implied bc you are in the class. Also, for future reference, I would try to avoid pointers because they don't really benefit in any way... they are just kinda there. No need to change, just keep it in mind.